libs/feature/search/src/lib/favorites/favorite-star/favorite-star.component.ts

```diff
@@ -1,21 +1,35 @@
 import {
+  AfterViewInit,
   ChangeDetectionStrategy,
   ChangeDetectorRef,
   Component,
+  ElementRef,
+  Inject,
+  InjectionToken,
   Input,
+  OnDestroy,
+  Optional,
+  ViewChild,
 } from '@angular/core'
 import { FavoritesService } from '../favorites.service'
 import { MetadataRecord } from '@geonetwork-ui/util/shared'
 import { map } from 'rxjs/operators'
 import { AuthService } from '@geonetwork-ui/feature/auth'
+import tippy from 'tippy.js'
+import { TranslateService } from '@ngx-translate/core'
+import { StarToggleComponent } from '@geonetwork-ui/ui/inputs'
+import { Subscription } from 'rxjs'
+import { LANG_2_TO_3_MAPPER } from '@geonetwork-ui/util/i18n'
+
+export const LOGIN_URL = new InjectionToken<string>('loginUrl')
```

Comment:
The default GN4 login URL should be the default value of this token. You can make token default values depend on injected services with the `inject` function: https://angular.io/api/core/InjectionToken#tree-shakable-injectiontoken

Reply:
Also I think this token should go somewhere else; it feels quite generic; IMO it should go in `feature/auth`

Reply:
I totally agree and moved the URL logic to the auth service. Couldn't quite get the default value of the token working, but the service decides whether to use the token or a default value.